gimme_aws_creds/registered_authenticators.py

```diff
@@ -0,0 +1,60 @@
+import hashlib
+import os
+import sqlite3
+
+
+# noinspection SqlDialectInspection,SqlNoDataSourceInspection
+class RegisteredAuthenticators(object):
+    """
+       The RegisteredAuthenticators Class manages a sqlite DB for gimme-aws-creds registered
+       FIDO authenticators.
+
+       There's a single table: registered authenticators, with two columns:
+       - cred_id_hash - sha512 of the registered credential id
+       - user - a user identifier (email, name, uid, ...)
+    """
+
+    DB_PATH_ENV_VAR = 'OKTA_REGISTERED_AUTHENTICATORS_DB'
+
+    def __init__(self, gac_ui):
+        """
+        :type gac_ui: ui.UserInterface
+        """
+        self.ui = gac_ui
+        self._db_path = self.ui.environ.get(self.DB_PATH_ENV_VAR,
+                                            os.path.join(self.ui.HOME, '.okta_aws_registered_authenticators'))
+        self._con = sqlite3.connect(self._db_path)
```

Comment:
Where is this database being created?  Is it really necessary that we have a separate database?

Reply:
Hey @Sector95 - thanks for the prompt review !

This small sqlite db is created along side the config file (ui-home, '.okta_aws_registered_authenticators');

An external store which saves a mapping of hashed credential id -> user id is needed in order to showcase
an appropriate name in the webauthn factor list.

Specifically, such a mapping is needed because okta has a predefined list of hard-coded authenticator ids (a-la Touch ID) for which it returns an appropriate name; For all other authenticators it returns a blank name, leading to a very confusing list of 'webauthn: webauthn' factors.

The reason I chose sqlite vs another file is for its consistency & atomicity, in the past, it proved to be very handy for saving small items of tabular information, handling fsync-s when appropriate, and alike.

Let me know what you think about it, thanks again !

Reply:
Are there a lot of these entries? Is there a reason something like a JSON blob in the configuration file couldn't be used for this?  A SQL database feels a bit like a jackhammer when a chisel would do.

Reply:
@Sector95 - Sounds good; I've migrated the code to use a json file instead.

I do think it's better to keep it separate from the current config due to several reasons:
- It's more of a repository than a configuration users needs to be aware of, or tweak
- It's relevant cross-profiles
- It's a bit dominant, due to the size of the sha512, and so it could clutter the config file if it were to sit there

Let me know what you think once you have a chance to glance on it.
Thanks again, have a great week ahead !